docs/specification.md

```diff
@@ -311,89 +432,153 @@ available state entry is `D`:
+example, one or more commits may have been pushed to a branch that do not meet
+gittuf policy. The repository is updated such that these commits are neutralized
+and all Git refs match their latest RSL entries. This can take two forms:
+
+1. The rejected commit is removed and the state of the repo is set to the prior
+commit which is known to be good. This is used when all rejected commits are
+together at the end of the commit graph, making it easy to remove all of them.
+
+2. The rejected commit is _reverted_ where a new commit is introduced that
+reverses all the changes made in the reverted commit. This is needed when "good"
+commits that must be retained are interspersed with "bad" commits that must be
+rejected.
+
+In both cases, new RSL entries and annotations may be used to record the
+incident or to skip the invalid RSL entries corresponding to the rejected
+changes. TODO: should these new RSL entries come from authorized users for the
+affected namespace?
+
+#### M2: Create RSL entry on behalf of another user
+
+This mechanism is necessary for adoptions where a subset of developers do not
+use gittuf. When they submit changes to the main copy of the repository, they do
+not include RSL entries. Therefore, when a change is pushed to a branch by a
+non-gittuf user A, a gittuf user B can submit an RSL entry on their behalf.
+Additionally, the entry must identify the original user (TODO: cross-reference
+with signed pushes?) and include some evidence about why B thinks the change
+came from A (TODO: signed push can be evidence option?).
+
+TODO: decide if B needs to be authorized for the affected branch to sign for A.
+
+### Recovery Scenarios
+
+These scenarios are some examples where recovery is necessary.
+
+#### A change is made without an RSL entry
+
+Bob does not use gittuf and pushes to a branch. Alice notices this as her gittuf
+client detects a push to the branch without an accompanying RSL entry. She
+validates that the change came from Bob and creates an RSL entry on his behalf,
+identifying him and including information about how she verified it was him.
+Alice applies M2.
+
+#### An incorrect RSL entry is added
+
+There are several ways in which an RSL entry can be considered "incorrect". If
+an entry is malformed (structurally), Git may catch it if it's not a valid
+commit. In such instances, the push from a buggy client is rejected altogether,
+meaning other users are not exposed to the malformed commit.
+
+Invalid entries that are not rejected by Git must be caught by gittuf. Some
+examples of such invalid entries are:
+
+* RSL entry is for a non-existing Git reference
+* Commit recorded in RSL entry does not exist
+* Commit recorded in RSL entry does not match the tip of the corresponding Git
+  reference
+* RSL annotation contains references to RSL entries that do not exist or are not
+  RSL entries (i.e. the annotation points to other commits in the repository)
```

Comment:
How does the next person that pulls always correctly know the last entry (head) of the RSL?

Reply:
Can you elaborate on "correctly"?

Reply:
I mean, how do I find the head of the RSL?  Can I be wrong / confused in some way?  How?

Reply:
Currently, the head of the RSL is just requested from `refs/gittuf/reference-state-log`. This opens the possibility that an attacker who controls that copy of the repo can freeze or set HEAD to an older RSL commit (rollback). This is in part discussed in the forking attack section as at least one developer knows the RSL's true state. In addition, we have to talk about whether we're introducing snapshot and timestamp roles.

If we go that direction, I think the snapshot role only needs to record the HEAD of the RSL, as the targets roles can be derived from the latest RSL entry for the policy namespace. In this case, the snapshot and timestamp would not be stored with the rest of the TUF metadata, but could still be in the repo, say `refs/gittuf/snapshot`.

Reply:
Actually, only timestamp is enough if it directly records the RSL head. The RSL has snapshot properties already.

Reply:
Okay, this is interesting.  I do think that we should consider using timestamp in this way.  It's also worth thinking about who actually holds that key and signs in this case though.  Why do we trust them and how is this done if the repo isn't gittuf enabled?

Reply:
TODO: Note how RSL replaces the snapshot role and how timestamp can be used to record the tip of the RSL.

Reply:
From IRL meeting: including timestamp has very mild benefits, we should leave it out. AI: discuss snapshot <-> RSL properties.

Reply:
Added some text to the RSL subsec above re its snapshot properties.